src/Command/Service/CreateCoreArchCommandService.php

```diff
@@ -0,0 +1,310 @@
+<?php
+
+/*
+ * Copyright 2005 - 2023 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Centreon\Command\Service;
+
+use Centreon\Command\CreateCoreArchCommand;
+use Centreon\Command\Model\FileTemplate;
+use Centreon\Command\Model\ModelTemplate\ModelTemplate;
+use Centreon\Command\Model\RepositoryTemplate\RepositoryInterfaceTemplate;
+use Centreon\Command\Model\UnitTestTemplate\UnitTestTemplate;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\{ChoiceQuestion, ConfirmationQuestion, Question};
+
+class CreateCoreArchCommandService
+{
+    private const REPO_PREFIX = 'centreon/centreon';
+
+    private RepositoryInterfaceTemplate $repositoryInterfaceTemplate;
+
+    /**
+     * @param string $srcPath
+     */
+    public function __construct(private string $srcPath)
+    {
+    }
+
+    /**
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     * @param mixed $questionHelper
+     *
+     * @return string
+     */
+    public function askForUseCaseType(InputInterface $input, OutputInterface $output, $questionHelper): string
```

Comment:
Stricter type here + changes accordingly
```suggestion
     * @param QuestionHelper $questionHelper
     *
     * @return string
     */
    public function askForUseCaseType(InputInterface $input, OutputInterface $output, QuestionHelper $questionHelper): string
```